mchess-master/mchess-master/CMoveList.h

```diff
@@ -30,20 +31,34 @@ class CMoveList
             m_moveList.push_back(move);
         }
 
+        void push_back_capture(const CMove& move)
+        {
+            m_moveList_capture.push_back(move);
+        }
+
         void clear()
         {
             m_moveList.clear();
+            m_moveList_capture.clear();
+
         }
 
         unsigned int size() const
         {
             return m_moveList.size();
+
+        }
+        unsigned int size_capture() const
+        {
+            return m_moveList_capture.size();
         }
 
         const CMove & operator [] (unsigned int ix) const { return m_moveList[ix]; }
+        const CMove get_capture(int ix) { return m_moveList_capture[ix]; }
```

Comment:
you should swap this

i.e [] returns capture
make a method return non_capture